net/gsmtp/gsmtp.go

```diff
@@ -38,35 +38,36 @@ func New(address, username, password string) *SMTP {
 // and then sends an email from address from, to addresses to, with
 // message msg.
 func (s *SMTP) SendMail(from, tos, subject, body string, contentType ...string) error {
-	if s.Address == "" {
-		return fmt.Errorf("address is necessary")
-	}
 
 	hp := strings.Split(s.Address, ":")
-	if len(hp) != 2 {
-		return fmt.Errorf("address format error")
+	if len(hp) == 1 {
+		server := s.Address
+		address := server + ":25"
+	} else if len(hp) == 2 {
+		server := hp[0]
+		address := s.Address
+	} else {
+		return fmt.Errorf("address is either empty or incorrect")
 	}
 
+	tosArr := []string
 	arr := strings.Split(tos, ";")
-	count := len(arr)
-	safeArr := make([]string, 0, count)
-	for i := 0; i < count; i++ {
-		if arr[i] == "" {
-			continue
+	for _, to := range arr {
+		// TODO: replace with regex
+		if strings.Containts(to, "@") {
+			tosArr = append(tosArr, to)
 		}
-		safeArr = append(safeArr, arr[i])
 	}
 
```

Comment:
using range instead of for i:=0 which is more Go style
also renamed safeArr into tosArr as it will be used later